--- a/src/main/java/org/bukkit/craftbukkit/Main.java
+++ b/src/main/java/org/bukkit/craftbukkit/Main.java
@@ -169,6 +_,14 @@
                         .ofType(String.class)
                         .defaultsTo("Unknown Server")
                         .describedAs("Name");
+
+                // Moose start
+                acceptsAll(asList("M", "moose-settings"), "File for moose settings")
+                    .withRequiredArg()
+                    .ofType(File.class)
+                    .defaultsTo(new File("moose.yml"))
+                    .describedAs("Yml file");
+                // Moose end
             }
         };
 
