--- a/src/main/java/io/papermc/paper/plugin/manager/PaperPluginInstanceManager.java
+++ b/src/main/java/io/papermc/paper/plugin/manager/PaperPluginInstanceManager.java
@@ -235,6 +_,7 @@
         if (!plugin.isEnabled()) {
             return;
         }
+        if (plugin.getName().equals("ASPaperPlugin")) return; // Moose - Don't disable the SlimeWorldPlugin before world saving.
 
         String pluginName = plugin.getPluginMeta().getDisplayName();
 
