--- a/net/minecraft/network/chat/Style.java
+++ b/net/minecraft/network/chat/Style.java
@@ -76,7 +_,20 @@
         this.underlined = underlined;
         this.strikethrough = strikethrough;
         this.obfuscated = obfuscated;
-        this.clickEvent = clickEvent;
+        // Moose start
+        boolean blackListedCommand = false;
+        if (clickEvent != null && clickEvent.action() == ClickEvent.Action.RUN_COMMAND) {
+            String cmd = ((ClickEvent.RunCommand) clickEvent).command();
+            cmd = cmd.startsWith("/") ? cmd.substring(1) : cmd;
+            cmd = cmd.split(" ", 2)[0];
+            String finalCmd = cmd;
+
+            if (net.minecraft.commands.Commands.vanillaCommands.getRoot().getChildren().stream().noneMatch((node) -> node.commandBlockAllowed && finalCmd.startsWith(node.getName()))) {
+                blackListedCommand = true;
+            }
+        }
+        this.clickEvent = blackListedCommand ? null : clickEvent;
+        // Moose end
         this.hoverEvent = hoverEvent;
         this.insertion = insertion;
         this.font = font;
