From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Legitimoose <legitimoose@gmail.com>
Date: Tue, 16 Jan 2024 13:46:19 -0800
Subject: [PATCH] Build changes


diff --git a/build.gradle.kts b/build.gradle.kts
index 3a1cad6046387bacf0e7ef83be8ee2e1a92137b7..e14cc0472c4ca267b03b75d27cbd2b44f0f6a8bf 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -3,6 +3,6 @@ plugins {
 
 dependencies {
     compileOnly(project(":api"))
-    compileOnly(project(":slimeworldmanager-api"))
+    compileOnly(project(":legitslimepaper-api")) // Moose
     implementation("com.github.luben:zstd-jni:1.5.2-2")
 }
