--- a/net/minecraft/world/level/BaseCommandBlock.java
+++ b/net/minecraft/world/level/BaseCommandBlock.java
@@ -104,7 +_,7 @@
         } else {
             this.successCount = 0;
             MinecraftServer server = this.getLevel().getServer();
-            if (server.isCommandBlockEnabled() && !StringUtil.isNullOrEmpty(this.command)) {
+            if (server.isCommandBlockEnabled() && com.legitimoose.lsp.MooseConfig.allowCommandBlocks && !StringUtil.isNullOrEmpty(this.command)) {
                 try {
                     this.lastOutput = null;
 
