--- a/net/minecraft/world/level/BaseCommandBlock.java
+++ b/net/minecraft/world/level/BaseCommandBlock.java
@@ -99,7 +_,7 @@
             return true;
         } else {
             this.successCount = 0;
-            if (level.isCommandBlockEnabled() && !StringUtil.isNullOrEmpty(this.command)) {
+            if (level.isCommandBlockEnabled() && com.legitimoose.lsp.MooseConfig.allowCommandBlocks && !StringUtil.isNullOrEmpty(this.command)) { // Moose - allow command blocks config option
                 try {
                     this.lastOutput = null;
 
