--- a/net/minecraft/world/level/ServerExplosion.java
+++ b/net/minecraft/world/level/ServerExplosion.java
@@ -319,7 +_,7 @@
     ) {
         this.level = level;
         this.source = source;
-        this.radius = (float) Math.max(radius, 0.0); // CraftBukkit - clamp bad values
+        this.radius = radius; // Moose - Remove value clamp
         this.center = center;
         this.fire = fire;
         this.blockInteraction = blockInteraction;
@@ -648,11 +_,7 @@
     }
 
     public void explode() {
-        // CraftBukkit start
-        if (this.radius < 0.1F) {
-            return;
-        }
-        // CraftBukkit end
+        // Moose - Remove minimum radius
         // Paper start - collision optimisations
         this.blockCache = new it.unimi.dsi.fastutil.longs.Long2ObjectOpenHashMap<>();
         this.chunkPosCache = new long[CHUNK_CACHE_WIDTH * CHUNK_CACHE_WIDTH];
