--- a/net/minecraft/server/Main.java
+++ b/net/minecraft/server/Main.java
@@ -1,6 +_,6 @@
 package net.minecraft.server;
 
-import com.mojang.authlib.yggdrasil.YggdrasilAuthenticationService;
+import com.legitimoose.lsp.MooseConfig;
 import com.mojang.datafixers.DataFixer;
 import com.mojang.logging.LogUtils;
 import com.mojang.serialization.Dynamic;
@@ -137,6 +_,8 @@
                 return;
             }
 
+            MooseConfig.init((File) optionSet.valueOf("moose-settings")); // Moose - Load moose.yml
+
             // Spigot start
             boolean eulaAgreed = Boolean.getBoolean("com.mojang.eula.agree");
             if (eulaAgreed) {
@@ -226,7 +_,7 @@
                 LOGGER.warn("Safe mode active, only vanilla datapack will be loaded");
             }
 
-            PackRepository packRepository = ServerPacksSource.createPackRepository(levelStorageAccess);
+            PackRepository packRepository = hasOptionSpec ? ServerPacksSource.createVanillaTrustedRepository() : ServerPacksSource.createPackRepository(levelStorageAccess); // Moose
             // CraftBukkit start
             File bukkitDataPackFolder = new File(levelStorageAccess.getLevelPath(net.minecraft.world.level.storage.LevelResource.DATAPACK_DIR).toFile(), "bukkit");
             if (!bukkitDataPackFolder.exists()) {
