--- a/net/minecraft/server/ServerFunctionLibrary.java
+++ b/net/minecraft/server/ServerFunctionLibrary.java
@@ -85,7 +_,7 @@
                     Identifier identifier1 = LISTER.fileToId(identifier);
                     map1.put(identifier1, CompletableFuture.supplyAsync(() -> {
                         List<String> lines = readLines(entry.getValue());
-                        return CommandFunction.fromLines(identifier1, this.dispatcher, commandSourceStack, lines);
+                        return CommandFunction.fromLines(identifier1, Commands.vanillaCommands, commandSourceStack, lines);
                     }, backgroundExecutor));
                 }
 
