--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -1349,7 +_,7 @@
     @Override
     public void handleEntityTagQuery(ServerboundEntityTagQueryPacket packet) {
         PacketUtils.ensureRunningOnSameThread(packet, this, this.player.level());
-        if (this.player.hasPermissions(2)) {
+        if (this.player.hasPermissions(2) || this.player.getBukkitEntity().hasPermission("minecraft.nbt.query")) { // Moose - Add nbt.query permission
             Entity entity = this.player.level().getEntity(packet.getEntityId());
             if (entity != null) {
                 try (ProblemReporter.ScopedCollector scopedCollector = new ProblemReporter.ScopedCollector(entity.problemPath(), LOGGER)) {
@@ -1375,7 +_,7 @@
     @Override
     public void handleBlockEntityTagQuery(ServerboundBlockEntityTagQueryPacket packet) {
         PacketUtils.ensureRunningOnSameThread(packet, this, this.player.level());
-        if (this.player.hasPermissions(2)) {
+        if (this.player.hasPermissions(2) || this.player.getBukkitEntity().hasPermission("minecraft.nbt.query")) { // Moose - Add nbt.query permission
             BlockEntity blockEntity = this.player.level().getBlockEntity(packet.getPos());
             CompoundTag compoundTag = blockEntity != null ? blockEntity.saveWithoutMetadata(this.player.registryAccess()) : null;
             this.send(new ClientboundTagQueryPacket(packet.getTransactionId(), compoundTag));
