From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Legitimoose <legitimoose@gmail.com>
Date: Tue, 16 Jan 2024 13:46:19 -0800
Subject: [PATCH] Build changes


diff --git a/build.gradle.kts b/build.gradle.kts
index 106fd711c9094e119ba99cc46de4601247e5985f..a134750eb548895b07ef8293a3bc61294ffd0a67 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -13,8 +13,8 @@ configurations.named(log4jPlugins.compileClasspathConfigurationName) {
 val alsoShade: Configuration by configurations.creating
 
 dependencies {
+    implementation(project(":legitslimepaper-api")) // Moose
     // ASWM start
-    implementation(project(":slimeworldmanager-api"))
     implementation(project(":core"))
     implementation("io.papermc.paper:paper-mojangapi:1.20-R0.1-SNAPSHOT") {
         exclude("io.papermc.paper", "paper-api")
