From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Legitimoose <legitimoose@gmail.com>
Date: Tue, 16 Jan 2024 13:46:19 -0800
Subject: [PATCH] Build changes


diff --git a/build.gradle.kts b/build.gradle.kts
index 106fd711c9094e119ba99cc46de4601247e5985f..49594f5772863eabca85a823980ba08e2cceb877 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -13,8 +13,8 @@ configurations.named(log4jPlugins.compileClasspathConfigurationName) {
 val alsoShade: Configuration by configurations.creating
 
 dependencies {
+    implementation(project(":legitslimepaper-api")) // Moose
     // ASWM start
-    implementation(project(":slimeworldmanager-api"))
     implementation(project(":core"))
     implementation("io.papermc.paper:paper-mojangapi:1.20-R0.1-SNAPSHOT") {
         exclude("io.papermc.paper", "paper-api")
@@ -75,7 +75,7 @@ tasks.jar {
         attributes(
             "Main-Class" to "org.bukkit.craftbukkit.Main",
             "Implementation-Title" to "CraftBukkit",
-            "Implementation-Version" to "git-SlimeWorldManager-$implementationVersion",
+            "Implementation-Version" to "git-LegitSlimePaper-$implementationVersion",
             "Implementation-Vendor" to date, // Paper
             "Specification-Title" to "Bukkit",
             "Specification-Version" to project.version,
@@ -159,7 +159,7 @@ fun TaskContainer.registerRunTask(
     name: String,
     block: JavaExec.() -> Unit
 ): TaskProvider<JavaExec> = register<JavaExec>(name) {
-    group = "slimeworldmanager"
+    group = "legitslimepaper"
     mainClass.set("org.bukkit.craftbukkit.Main")
     standardInput = System.`in`
     workingDir = rootProject.layout.projectDirectory
diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index 91b318e50fb271bd836f891a30270601f5a94790..33ba5fafba15d5775880666e41d9e9ed187a2dfa 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -1712,7 +1712,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
 
     @DontObfuscate
     public String getServerModName() {
-        return "SlimeWorldManager"; // ASWM // Paper - Paper > // Spigot - Spigot > // CraftBukkit - cb > vanilla!
+        return "LegitSlimePaper"; // Moose // ASWM // Paper - Paper > // Spigot - Spigot > // CraftBukkit - cb > vanilla!
     }
 
     public SystemReport fillSystemReport(SystemReport details) {
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index f824b6a4e17bed3e82c25f7c6711196f0bb52cd3..5999d17db3f96deb6a04b87d0c116f9370a9ec00 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -269,7 +269,7 @@ import javax.annotation.Nullable; // Paper
 import javax.annotation.Nonnull; // Paper
 
 public final class CraftServer implements Server {
-    private final String serverName = "SlimeWorldManager"; // Paper // SWM
+    private final String serverName = "LegitSlimePaper"; // Paper // SWM // Moose
     private final String serverVersion;
     private final String bukkitVersion = Versioning.getBukkitVersion();
     private final Logger logger = Logger.getLogger("Minecraft");
diff --git a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
index 29e29fb2228f08dff3edd8bf651220b83b16ed7b..d0473ffb840c8668719f8b6e3200db615c4a3277 100644
--- a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
@@ -11,7 +11,7 @@ public final class Versioning {
     public static String getBukkitVersion() {
         String result = "Unknown-Version";
 
-        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/com.infernalsuite.aswm/slimeworldmanager-api/pom.properties"); // ASWM
+        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/com.legitimoose.legitslimepaper/legitslimepaper-api/pom.properties"); // ASWM // Moose
         Properties properties = new Properties();
 
         if (stream != null) {
