From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Legitimoose <legitimoose@gmail.com>
Date: Tue, 16 Jan 2024 13:46:19 -0800
Subject: [PATCH] Build changes


diff --git a/build.gradle.kts b/build.gradle.kts
index 316d03791a97af50f882c0271cbacf3d2a7af5b3..aaeeb2920d44356820d0708d4830aa4a1a324674 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -14,8 +14,8 @@ configurations.named(log4jPlugins.compileClasspathConfigurationName) {
 val alsoShade: Configuration by configurations.creating
 
 dependencies {
+    implementation(project(":legitslimepaper-api")) // Moose
     // ASWM start
-    implementation(project(":slimeworldmanager-api"))
     implementation(project(":core"))
     // ASWM end
     // Paper start
@@ -84,14 +84,14 @@ tasks.jar {
         val gitBranch = git("rev-parse", "--abbrev-ref", "HEAD").getText().trim() // Paper
         attributes(
             "Main-Class" to "org.bukkit.craftbukkit.Main",
-            "Implementation-Title" to "AdvancedSlimePaper",
+            "Implementation-Title" to "LegitSlimePaper", // Moose
             "Implementation-Version" to implementationVersion,
             "Implementation-Vendor" to date, // Paper
-            "Specification-Title" to "AdvancedSlimePaper",
+            "Specification-Title" to "LegitSlimePaper", // Moose
             "Specification-Version" to project.version,
-            "Specification-Vendor" to "InfernalSuite Team",
-            "Brand-Id" to "infernalsuite:advancedslimepaper",
-            "Brand-Name" to "AdvancedSlimePaper",
+            "Specification-Vendor" to "Legitimoose",
+            "Brand-Id" to "legitimoose:legitslimepaper",
+            "Brand-Name" to "LegitSlimePaper",
             "Build-Number" to (build ?: ""),
             "Build-Time" to Instant.now().toString(),
             "Git-Branch" to gitBranch, // Paper
@@ -148,7 +148,7 @@ fun TaskContainer.registerRunTask(
     name: String,
     block: JavaExec.() -> Unit
 ): TaskProvider<JavaExec> = register<JavaExec>(name) {
-    group = "slimeworldmanager"
+    group = "legitslimepaper"
     mainClass.set("org.bukkit.craftbukkit.Main")
     standardInput = System.`in`
     workingDir = rootProject.layout.projectDirectory
diff --git a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
index 29e29fb2228f08dff3edd8bf651220b83b16ed7b..d0473ffb840c8668719f8b6e3200db615c4a3277 100644
--- a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
@@ -11,7 +11,7 @@ public final class Versioning {
     public static String getBukkitVersion() {
         String result = "Unknown-Version";
 
-        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/com.infernalsuite.aswm/slimeworldmanager-api/pom.properties"); // ASWM
+        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/com.legitimoose.legitslimepaper/legitslimepaper-api/pom.properties"); // ASWM // Moose
         Properties properties = new Properties();
 
         if (stream != null) {
