From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Legitimoose <legitimoose@gmail.com>
Date: Thu, 18 Jan 2024 17:15:40 -0800
Subject: [PATCH] Allow negative explosion radius


Co-authored-by: Nico314159 <nicolino.will@gmail.com>

diff --git a/src/main/java/net/minecraft/world/level/Explosion.java b/src/main/java/net/minecraft/world/level/Explosion.java
index 45243249a561440512ef2a620c60b02e159c80e2..e416f2909a594b737d2228bc6e9b1087ae84724c 100644
--- a/src/main/java/net/minecraft/world/level/Explosion.java
+++ b/src/main/java/net/minecraft/world/level/Explosion.java
@@ -87,7 +87,7 @@ public class Explosion {
         this.hitPlayers = Maps.newHashMap();
         this.level = world;
         this.source = entity;
-        this.radius = (float) Math.max(power, 0.0); // CraftBukkit - clamp bad values
+        this.radius = (float) power; // Moose - Remove value clamp
         this.x = x;
         this.y = y;
         this.z = z;
@@ -402,11 +402,7 @@ public class Explosion {
     }
 
     public void explode() {
-        // CraftBukkit start
-        if (this.radius < 0.1F) {
-            return;
-        }
-        // CraftBukkit end
+        // Moose - Remove minimum radius
         this.level.gameEvent(this.source, GameEvent.EXPLODE, new Vec3(this.x, this.y, this.z));
         Set<BlockPos> set = Sets.newHashSet();
         boolean flag = true;
