From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Legitimoose <legitimoose@gmail.com>
Date: Thu, 18 Jan 2024 17:15:40 -0800
Subject: [PATCH] Allow negative explosion radius

Co-authored-by: Nico314159 <nicolino.will@gmail.com>

diff --git a/src/main/java/net/minecraft/world/level/Explosion.java b/src/main/java/net/minecraft/world/level/Explosion.java
index 082b804f4793f72e76361f5427f0358273454b3d..84c0d95b8a7991cdcb9b88cbe86749b58ab448a8 100644
--- a/src/main/java/net/minecraft/world/level/Explosion.java
+++ b/src/main/java/net/minecraft/world/level/Explosion.java
@@ -98,7 +98,7 @@ public class Explosion {
         this.hitPlayers = Maps.newHashMap();
         this.level = world;
         this.source = entity;
-        this.radius = (float) Math.max(power, 0.0); // CraftBukkit - clamp bad values
+        this.radius = (float) power; // Moose - Remove value clamp
         this.x = x;
         this.y = y;
         this.z = z;
@@ -425,11 +425,7 @@ public class Explosion {
     }
 
     public void explode() {
-        // CraftBukkit start
-        if (this.radius < 0.1F) {
-            return;
-        }
-        // CraftBukkit end
+        // Moose - Remove minimum radius
         this.level.gameEvent(this.source, (Holder) GameEvent.EXPLODE, new Vec3(this.x, this.y, this.z));
         Set<BlockPos> set = Sets.newHashSet();
         boolean flag = true;
