From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Legitimoose <legitimoose@gmail.com>
Date: Thu, 18 Jan 2024 17:07:56 -0800
Subject: [PATCH] Add permission for game master blocks


diff --git a/src/main/java/net/minecraft/world/entity/player/Player.java b/src/main/java/net/minecraft/world/entity/player/Player.java
index 686374e89e41b0917791264f3281f7384835bca8..4605f8c0e2ecb22ca0524ccec6eca9c420965912 100644
--- a/src/main/java/net/minecraft/world/entity/player/Player.java
+++ b/src/main/java/net/minecraft/world/entity/player/Player.java
@@ -2260,7 +2260,7 @@ public abstract class Player extends LivingEntity {
     }
 
     public boolean canUseGameMasterBlocks() {
-        return this.abilities.instabuild && this.getPermissionLevel() >= 2;
+        return this.abilities.instabuild && (this.getPermissionLevel() >= 2  || this.getBukkitEntity().hasPermission("minecraft.commandblock")); // Moose - Add permission for interacting with GameMasterBlocks
     }
 
     @Override
