From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Legitimoose <legitimoose@gmail.com>
Date: Thu, 18 Jan 2024 17:15:40 -0800
Subject: [PATCH] Allow negative explosion radius

Co-authored-by: Nico314159 <nicolino.will@gmail.com>

diff --git a/src/main/java/net/minecraft/world/level/Explosion.java b/src/main/java/net/minecraft/world/level/Explosion.java
index f696afd7e241bf1966a2d505b5d59bff824b43e4..7b4582c6314e0719f9968947ada22e21f448a804 100644
--- a/src/main/java/net/minecraft/world/level/Explosion.java
+++ b/src/main/java/net/minecraft/world/level/Explosion.java
@@ -341,7 +341,7 @@ public class Explosion {
         this.hitPlayers = Maps.newHashMap();
         this.level = world;
         this.source = entity;
-        this.radius = (float) Math.max(power, 0.0); // CraftBukkit - clamp bad values
+        this.radius = (float) power; // Moose - Remove value clamp
         this.x = x;
         this.y = y;
         this.z = z;
@@ -403,11 +403,7 @@ public class Explosion {
     }
 
     public void explode() {
-        // CraftBukkit start
-        if (this.radius < 0.1F) {
-            return;
-        }
-        // CraftBukkit end
+        // Moose - Remove minimum radius
         this.level.gameEvent(this.source, (Holder) GameEvent.EXPLODE, new Vec3(this.x, this.y, this.z));
 
         // Paper start - collision optimisations
