From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: PolishKrowa <45701824+Mrredstone5230@users.noreply.github.com>
Date: Fri, 26 Jul 2024 23:26:31 -0400
Subject: [PATCH] Make /return only run allowed commands


diff --git a/src/main/java/net/minecraft/server/commands/ReturnCommand.java b/src/main/java/net/minecraft/server/commands/ReturnCommand.java
index 9f82ca1fee2a319d52a4106c3581f5e9a9554a9e..265654d583e20442fa64bbf1f793cbc04ed5d549 100644
--- a/src/main/java/net/minecraft/server/commands/ReturnCommand.java
+++ b/src/main/java/net/minecraft/server/commands/ReturnCommand.java
@@ -6,6 +6,9 @@ import com.mojang.brigadier.builder.LiteralArgumentBuilder;
 import com.mojang.brigadier.builder.RequiredArgumentBuilder;
 import com.mojang.brigadier.context.ContextChain;
 import java.util.List;
+
+import com.mojang.brigadier.tree.CommandNode; // Moose
+import net.minecraft.commands.Commands; // Moose
 import net.minecraft.commands.ExecutionCommandSource;
 import net.minecraft.commands.execution.ChainModifiers;
 import net.minecraft.commands.execution.CustomCommandExecutor;
@@ -21,11 +24,11 @@ public class ReturnCommand {
             (LiteralArgumentBuilder<T>)LiteralArgumentBuilder.<ExecutionCommandSource>literal("return")
                 .requires(source -> source.hasPermission(2))
                 .then(
-                    RequiredArgumentBuilder.<T, Integer>argument("value", IntegerArgumentType.integer())
-                        .executes(new ReturnCommand.ReturnValueCustomExecutor<>())
+                    RequiredArgumentBuilder.argument("value", IntegerArgumentType.integer())
+                        .executes(new ReturnCommand.ReturnValueCustomExecutor())
                 )
-                .then(LiteralArgumentBuilder.<T>literal("fail").executes(new ReturnCommand.ReturnFailCustomExecutor<>()))
-                .then(LiteralArgumentBuilder.<T>literal("run").forward(dispatcher.getRoot(), new ReturnCommand.ReturnFromCommandCustomModifier<>(), false))
+                .then(LiteralArgumentBuilder.literal("fail").executes(new ReturnCommand.ReturnFailCustomExecutor()))
+                .then(LiteralArgumentBuilder.<T>literal("run").forward((CommandNode<T>) Commands.vanillaCommands.getRoot(), new ReturnCommand.ReturnFromCommandCustomModifier<>(), false)) // Moose - Forward to vanilla/allowed commands
         );
     }
 
