--- a/net/minecraft/commands/functions/MacroFunction.java
+++ b/net/minecraft/commands/functions/MacroFunction.java
@@ -50,6 +_,7 @@
 
     @Override
     public InstantiatedFunction<T> instantiate(@Nullable CompoundTag arguments, CommandDispatcher<T> dispatcher) throws FunctionInstantiationException {
+        dispatcher = (CommandDispatcher<T>) net.minecraft.commands.Commands.vanillaCommands; // Moose
         if (arguments == null) {
             throw new FunctionInstantiationException(Component.translatable("commands.function.error.missing_arguments", Component.translationArg(this.id())));
         } else {
